### PR TITLE
[production/RRFS.v1] Modify run directory structure for DA processing tasks

### PR DESCRIPTION
**Pull request by BenjaminBlake-NOAA**
## DESCRIPTION OF CHANGES: 

- This PR for modifying the run directory structure is building off of PRs #348 and #356.  The process_bufr, process_smoke, process_radarref, process_lightning, process_pm, and process_glmfed tasks have been updated to use `${RUN}_${task}_${envir}_${cyc}` run directories.  After this PR, there will be one final PR to update all the remaining workflow tasks.
- The CYCLE_DIR (now DATAROOT) input argument has been removed from the process_pm, process_lightning, process_radarref, and process_bufr ex-scripts.  This input argument is not necessary because CYCLE_DIR is not referenced anywhere in these ex-scripts.
- Several miscellaneous fixes to typos/comments in the J-jobs/ex-scripts are also included (e.g. changing FV3 to RRFS).
- Note that CYCLE_DIR is still defined in FV3LAM_wflow.xml.  It hasn't been removed yet because the other tasks which need to be updated (e.g. JRRFS_RUN_ANALYSIS) still use it.
- One thing that I have not done yet and will need to do is update the NWGES directory structure, which also uses date/cycle-specific directories.  I think this change can be done separately from updating the run directory structure.

## TESTS CONDUCTED: 

- The process_bufr, process_radarref, and process_smoke tasks were successfully tested with these changes.  The process_pm, process_lightning, and process_glmfed tasks have not been tested because they are not turned on with the DA engineering test, but they will be tested with a full RRFS system test once all the other workflow tasks have been updated.

### Machines/Platforms:

- WCOSS2
  - [x] Cactus/Dogwood
  - [ ] Acorn
- RDHPCS
  - [ ] Hera
  - [ ] Jet
  - [ ] Orion
  - [ ] Hercules

### Test cases: 

- [ ] Engineering tests
  - [ ] Non-DA engineering test
  - [x] DA engineering test
    - [ ] Retro
    - [ ] Ensemble
    - [ ] Parallel
- [ ] RRFS fire weather
- [ ] RRFS_A:
- [ ] RRFS_B:
- [ ] RTMA:
- [ ] Others:

## ISSUE: 

- Continues to address issue #237

### CODE REVIEW

File: jobs/JRRFS_PROCESS_GLMFED
Comment (MatthewPyle-NOAA) [inline jobs/JRRFS_PROCESS_GLMFED:1]: I never have a good feel for error checking with python stuff, but would it make sense to have the typical error checking after calling the exrrfs_process_glmfed.py script?

export err=$?; err_chk

**Review by MatthewPyle-NOAA — COMMENTED** [review]
One small question, but overall looks like more progress on pushing things towards a production-like setup.  Thanks!

File: jobs/JRRFS_PROCESS_GLMFED
Comment (BenjaminBlake-NOAA) [reply jobs/JRRFS_PROCESS_GLMFED:1]: I also wasn't sure whether to include it.  But I'll go ahead and add it.  I think eventually we may want an intermediate ex-script for this task.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Thanks for that change...approving.

**Pull request merged**